### DISCUSSION TITLE
Feature Request: Please make @this.call return a promise

### DISCUSSION THREAD

**Discussion by randohinn** [livewire/livewire#1061] [4. General]
Hi.

I'm looking for a way to detect success of a `@this.call`. The easiest way would be for @this.call to be a promise, so one could chain `.then` to it. Thus this feature request to please make this return a promise. Did not find where this is implemented in the code, otherwise I would have PR'ed it in myself.

**Comment by ryatkins** [comment]
From the docs, would this work?

```
// Calling a Livewire method and doing something with its result
$wire.someMethod(someParam)
    .then(result => { ... })
```

**Reply by randohinn** [reply]
Maybe.. I honestly don't even remember why or where I needed this... Must've gotten around it.
